barretenberg/cpp/docs/src/sumcheck-outline.md

```diff
@@ -0,0 +1,438 @@
+    \f}
+to establish that \f$ F(P_1(\vec \ell),\ldots, P_N(\vec \ell) ) = 0 \f$, i.e. that \f$ F \f$ is satisfied at every
+point of \f$\{0,1\}^d\f$.
+
+ In the implementation, the relation polynomial \f$ F \f$ is specified by the Flavor.
+
+ ### Main Parameters {#MainParameters}
+
+The following constants are used in this exposition. 
+
+ |     Notation      |           | \f$ \sim \f$   Upper Bound |
+ --------------------|---------------|-----------|
+ | \f$ d \f$         | \ref multivariate_d "number of variables" in multilinear  polynomials \f$ P_1,\ldots, P_N\f$        | \f$ 20 \f$  |
+ | \f$ N \f$         | number of Prover Polynomials specified by <Flavor ::NUM_ALL_ENTITIES>                               | \f$ 60 \f$  | 
+ | \f$ N_w \f$  	 | number of Witness Polynomials specified bt <Flavor ::NUM_WITNESS_ENTITIES>                          | \f$ 17 \f$  | 
+ | \f$ n \f$         | \ref multivariate_n "size of the hypercube", i.e. \f$ 2^d\f$.                                       | \f$ 2^{20} \f$ |
+ | \f$ D \f$         | \ref bb::SumcheckProverRound< Flavor >::BATCHED_RELATION_PARTIAL_LENGTH "maximum partial degree of" \f$\tilde{F}\f$ | \f$ 12 \f$ |
+ | \f$ D_w\f$        | maximum witness degree of \f$ F \f$ | \f$ 5 \f$ |
+
+#### Maximum Witness Degree 
+The significance of this parameter becomes apparent in Section [Masking Evaluations of Multilinear Witnesses](#MaskingEvalsOfWitnesses). It is formally defined as follows 
+\f{align}{
+	D_w = \deg_{P_1, \ldots, P_{N_w}} F(P_1,\ldots, P_{N})
+\f}
+where by \f$ \deg_{P_1, \ldots, P_{N_w}} \f$ we mean the <b> total degree </b> of relation polynomial \f$ F \f$ in the first \f$N_w\f$ variables. 
+
+## Sumcheck Prover Algorithm {#NonZKSumcheckProver}
+- - -
+We remark that prior to running Sumcheck, the prover commits to multilinear polynomials \f$P_1,\ldots, P_{N_w}\f$, and sends the commitments to the verifier and that the total sum and the relation polynomial \f$ \tilde F \f$ are public. 
+
+The prover algorithm is implemented in the \ref bb::SumcheckProver< Flavor > "Sumcheck Prover" class. See its documentation for a more detailed description of methods described below. The Sumcheck Round routine is abstracted into \ref bb::SumcheckProverRound< Flavor > "Sumcheck Prover Round" class, which contains most computational steps.
+
+
+
+####  Set up Prover Polynomials {#ProverPolynomialsSetup}
+
+The polynomials \f$P_1,\ldots, P_N\f$ are abstracted in the class ProverPolynomials specific to a Flavor, e.g. \ref bb::GoblinUltraFlavor::ProverPolynomials "Goblin Ultra Flavor". 
+Sumcheck Prover algorithm takes a reference to an object of this class. 
+
+####  Compute Round Univariates and add them to Transcript {#ComputeRoundUnivariates}
+The prover evaluates the round univariate 
+\f{align}{
+	\tilde{S}^i = \sum_{\vec \ell \in \{0,1\}^{d-1-i}} \tilde{F}\left(u_0,\ldots, u_{i-1}, X_i,\vec \ell\right)
+\f} 
+over the domain \f$ 0,\ldots, D \f$. 
```

Comment:
mention we use barycentric evaluation and we start from domain 0,1  and maybe why this is needed?

Reply:
added a link to the definition of T^i, seems fine to call them auxiliary univariates at this point.

Reply:
Expanded the explanation. I hope, I got your comment right